packages/teleterm/src/ui/ServerConnect/useServerConnect.ts

```diff
@@ -23,8 +23,10 @@ export default function useServerConnect({ serverUri, onClose }: Props) {
   const logins = cluster?.loggedInUser?.sshLoginsList || [];
 
   const connect = (login: string) => {
+    const rootCluster =
+      ctx.clustersService.findRootClusterByResource(serverUri);
     const documentsService = ctx.workspacesService.getWorkspaceDocumentService(
-      cluster.uri
+      rootCluster.uri
```

Comment:
Yeah, I'm still kinda conflicted whether we should convert a resource URI to a root cluster URI this way. The root cluster URI seems to be baked into every other URI, so theoretically we should be able to get it just from operations on that string alone, without having to use `clustersService`. My main concern is that in other places it'll require pulling in that `clustersService` which can unnecessarily complicate tests.

We might think about something like `routing.extractRootClusterUri` in the future. For now I added just `ensureRootClusterUri` there which takes any kind of cluster URI and converts it to a root cluster URI.

Reply:
Good point, I will add it to our master ticket